### DISCUSSION TITLE
How to unset git config?

### DISCUSSION THREAD

**Discussion by iriseel** [community/community#23397] [New to GitHub]
<p>After encountering authentication issues with pushing from Github Desktop, I followed some advice <a href="https://gist.github.com/taoyuan/bfa3ff87e4b5611b5cbe" rel="noopener nofollow ugc">here</a>, and ended up making some changes to my git config that I now cannot figure out how to undo.<br>
When I enter in “git config --global --edit” in Terminal, the popup window contains these lines, which are the ones I’d like to get rid of but can’t:</p>
<pre><code class="lang-auto">[url "git@github.com:"]
        insteadOf = https://github.com/
[url "https:github.com/"]
        insteadOf = git@github.com:
[url "git://"]
        insteadOf = https://
</code></pre>
<p>Could anyone help with this?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="iriseel" data-post="1" data-topic="188363">
<div class="title">
<div class="quote-controls"></div>
 iriseel:</div>
<blockquote>
<p>which are the ones I’d like to get rid of but can’t:</p>
</blockquote>
</aside>
<p>What do you mean by “can’t”? Deleting the relevant parts of the git config should work to get rid of settings. Alternatively you could look at <a href="https://git-scm.com/docs/git-config#Documentation/git-config.txt---unset"><code>git config --unset</code></a>.</p>

**Comment by iriseel** [comment]
<p>Hi airtower-luna, thanks for your reply! By “can’t” I mean that when I hit delete on any of the letters, it just moves my typing position back like a left-arrow, without erasing any actual characters. I can replace characters by typing letters, but not delete them.</p>
<p>However, I took another look at your unset rec and figured out how to undo my git config mistakes!<br>
For anyone who’s running into similar issues, I ended up inputting the following code in Terminal:</p>
<pre><code class="lang-auto">git config --global --unset-all url.git@github.com:.insteadof
git config --global --unset-all url.https:github.com/.insteadof
git config --global --unset-all url.git://.insteadof
</code></pre>
<p>So yeah, still not sure why I couldn’t delete characters in “git config --global --edit” directly, but at least I managed to unset them outside. Thanks again!</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="iriseel" data-post="4" data-topic="188363">
<div class="title">
<div class="quote-controls"></div>
 iriseel:</div>
<blockquote>
<p>still not sure why I couldn’t delete characters in “git config --global --edit” directly</p>
</blockquote>
</aside>
<p>Is the editor that was started with the <code>--edit</code> option by any chance Vi? That one is known for its obscure keyboard commands, but you can find guides online. :sweat_smile:</p>

**Comment by iriseel** [comment]
<p>Nah, it’s just Terminal, weirdly. Deleting has worked everywhere else in Terminal except for in that “git config --global --edit” window :confused:</p>

**Comment by airtower-luna** [comment]
<p>Vi would usually run in the terminal (though GUI versions exist). It’s a very common default editor on Unix-ish systems, and if you’re using Windows I wouldn’t be surprised at all if Git Bash used it, too.</p>

**Comment by iriseel** [comment]
<p>Ohh that would make sense then! I’m using mac but I just looked it up online and Vi seems to come preinstalled on most macOS.</p>

**Comment by parshOne** [comment]
I have query that in my azure devops pipeline agent, repository is being checked out but even after executing git config --unset-all <url_repo> it still giving warning in next line saying unable to remove<url> git config. To remvoe the creadetial do git config --unset-all so why it's happening
